nilearn/regions/tests/test_region_extractor.py

```diff
@@ -27,38 +36,69 @@ def _make_random_data(shape):
     return img, data
 
 
-def test_invalid_thresholds_in_threshold_maps_ratio():
-    maps, _ = generate_maps((10, 11, 12), n_regions=2)
+@pytest.fixture(scope="module")
+def dummy_map():
+    """Generate a small dummy map.
+
+    Use for error testing
+    """
+    maps, _ = generate_maps(shape=(6, 6, 6), n_regions=N_REGIONS)
+    return maps
+
+
+@pytest.fixture
+def labels_img():
+    shape, affine = _shape_and_affine_labeled_regions()
+    n_regions = 9  # DO NOT CHANGE (some tests expect this value)
+    return generate_labeled_regions(shape, affine=affine, n_regions=n_regions)
+
+
+@pytest.fixture
+def maps():
+    maps, _ = generate_maps(shape=MAP_SHAPE, n_regions=N_REGIONS)
+    return maps
 
-    for invalid_threshold in ["80%", "auto", -1.0]:
-        with pytest.raises(
-            ValueError,
-            match="threshold given as ratio to the number of voxels must "
-            "be Real number and should be positive "
-            "and between 0 and total number of maps "
-            f"i.e. n_maps={maps.shape[-1]}. "
-            f"You provided {invalid_threshold}",
-        ):
-            _threshold_maps_ratio(maps, threshold=invalid_threshold)
+
+@pytest.fixture
+def maps_and_mask():
+    maps, mask_img = generate_maps(shape=MAP_SHAPE, n_regions=N_REGIONS)
+    return maps, mask_img
+
+
+@pytest.fixture
+def map_img_3D():
+    rng = np.random.RandomState(42)
+    map_img = np.zeros(MAP_SHAPE) + 0.1 * rng.standard_normal(size=MAP_SHAPE)
+    return nibabel.Nifti1Image(map_img, affine=np.eye(4))
+
+
+@pytest.mark.parametrize("invalid_threshold", ["80%", "auto", -1.0])
+def test_invalid_thresholds_in_threshold_maps_ratio(
+    dummy_map, invalid_threshold
+):
+    with pytest.raises(
+        ValueError,
+        match="threshold given as ratio to the number of voxels must "
+        "be Real number and should be positive "
+        "and between 0 and total number of maps "
+        f"i.e. n_maps={dummy_map.shape[-1]}. "
+        f"You provided {invalid_threshold}",
+    ):
+        _threshold_maps_ratio(maps_img=dummy_map, threshold=invalid_threshold)
 
 
-def test_nans_threshold_maps_ratio():
-    maps, _ = generate_maps((10, 10, 10), n_regions=2)
+def test_nans_threshold_maps_ratio(maps):
     data = get_data(maps)
     data[:, :, 0] = np.nan
 
     maps_img = nibabel.Nifti1Image(data, np.eye(4))
     _threshold_maps_ratio(maps_img, threshold=0.8)
 
 
-def test_threshold_maps_ratio():
+def test_threshold_maps_ratio(maps):
```

Comment:
Could you turn the comment just after this line into a docstring ?

Reply:
done